f8a_report/v2/report_generator.py

```diff
@@ -140,6 +142,11 @@ def analyse_stack(self, stacks_data, report_template) -> dict:
             stack = stack[0]
             stack_info_template = self.get_stack_info_template()
             ecosystem = stack.get('ecosystem')
+            if ecosystem not in self.supported_ecosystems:
+                logger.error("Unexpected Ecosystem type: %s, value: %s", type(ecosystem), ecosystem)
+                ecosystem = re.sub(r'\W+', '', ecosystem)
```

Comment:
Why are you correcting the ecosystem and then checking again for error?

Reply:
@prashbnair handling 2 cases here:
- when ecosystem= "golang,"-> Cleaning Ecosystem and then continuing process
- when ecosystem="golangg" or whatever -> Explicitly raising assert Error in First place to show it in logs

Reply:
I am not in favour of correcting and continuing, we will not know the errors. Let us  write the different ecosystems in the logs and continue. That way we can correct it.

Reply:
Probably I was not clear enough
- when ecosystem= "golang,"-> Cleaning Ecosystem and then continuing process (Logging is there)
- when ecosystem="golangg" or whatever -> Cleaning is not possible, then Explicitly raising assert Error and show it in logs and stopping Process at very moment.

Reply:
Lets us not fix anything here, why should we fix just one mistake, I feel it is not required and we need to fix the data corruption issue. Until then, let it continue to fail.

Reply:
Ack. Closing PR